src/src/index.ts

```diff
@@ -154,13 +155,45 @@ export function urlEncode(){
   setText(encodeURIComponent(getText()));
 }
 
-export function formatCertificate() {
-  var cert = getText();
-  if (!cert) {
+export async function extractCertificate() {
+  const workspace = vscode.workspace;
+  if (!workspace) {
     return;
   }
 
-  setText(utils.certToPEM(cert));
+  var xmlText = getText();
+  if (!xmlText) {
+    return;
+  }
+  var samlElement = createSAMLElement(xmlText);
+  if (!(samlElement instanceof BaseElement)) {
+    return vscode.window.showErrorMessage("The text doesn't appear to be a valid SAML XML document");
+  }
+  const certs = samlElement.extractCertificates();
+  if (certs.length === 0) {
+    return vscode.window.showErrorMessage("Couldn't find an x509 certificate");
+  }
+  let cert;
+  // if there's just one, or they are all the same, don't bother showing the options
+  if (certs.length === 1 || certs.every(cert => cert.text === certs[0].text)) {
+    cert = certs[0];
+  } else {
+    const selection = await vscode.window.showQuickPick(certs.map(cert => cert.label), {
+      placeHolder: "Pick the certificate that you want to extract"
+    });
+    if (!selection) {
+      return;
+    }
+    cert = certs.find(cert => cert.label === selection);
+    if (!cert) {
+      return;
+    }
+  }
+  const textDocument = await workspace.openTextDocument( {
```

Comment:
@nicosabena I'm getting a compilation issue here when trying to publish it

<img width="852" alt="image" src="https://user-images.githubusercontent.com/1765468/200863881-470e7209-f7f8-411e-ad1f-63b26a81c1eb.png">